aspnet/fundamentals/file-providers.rst

```diff
@@ -1,8 +1,154 @@
+
+In this article's sample, a console application is configured to display a message whenever a text file is modified:
+
+.. literalinclude:: file-providers/sample/src/WatchConsole/Program.cs
+  :emphasize-lines: 11-12,24,26-27
+
+The result, after saving the file several times:
+
+.. image:: /fundamentals/file-providers/_static/watch-console.png
+
+.. note:: Some file systems, such as Docker containers and network chares, may not reliably send change notifications. Set the ``DOTNET_USE_POLLINGFILEWATCHER`` environment variable to ``1`` or ``true`` to poll the file system for changes every 4 seconds.
+
+Globbing Patterns
+-----------------
+
+File system paths use wilcard patterns called *globbing patterns*. These simple patterns can be used to specify groups of files. The two wildcard characters are ``*`` and ``**``.
+
+``*``
+    Matches anything at the current folder level, or any filename, or any file extension. Matches are terminated by ``/`` and ``.`` characters in the file path.
+
+``**``
+    Matches anything across multiple directory levels. Can be used to recursively match many files within a directory hierarchy.
+
+Globbing Pattern Examples
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``directory/file.txt``
+    Matches a specific file in a specific directory.
+
+``directory/*.txt``
+    Matches all files with ``.txt`` extension in a specific directory.
+
+``directory/*/project.json``
+    Matches all ``project.json`` files in directories exactly one level below the ``directory`` directory.
+
+``directory/**/*.txt``
+    Matches all files with ``.txt`` extension found anywhere under the ``directory`` directory.
+
+File Provider Usage
+-------------------
+
+Several parts of ASP.NET Core utilize file providers. ``IHostingEnvironment`` exposes the app's content root and web root as ``IFileProvider`` types. The static files middleware uses file providers to locate static files. Razor makes heavy use of ``IFileProvider`` in locating views. Dotnet's publish functionality uses file providers and globbing patterns to specify which files should be published.
+
+If your ASP.NET Core app requires file system access, you can request an instance of ``IFileProvider`` through dependency injection, and then use its methods to perform the access, as shown in this sample. This allows you to configure the provider once, when the app starts up, and reduces the number of implementation types your app instantiates.
```

Comment:
This paragraph sounds like an article summary but there's no Summary heading, it's under the FP Usage heading.